blogposts/2021/release-3.31.md

```diff
@@ -0,0 +1,77 @@
+---
+title: 'Sourcegraph 3.31 release'
+publishDate: 2021-09-01T10:00-07:00
+description: 'Sourcegraph 3.31 introduces the Private Beta release of Code Insights, complete with no repository limits plus repository-based filters. It also introduces automated migration of saved search notifications to code monitoring.'
+tags: [blog, release]
+slug: 'release/3.31'
+published: true
+heroImage: https://sourcegraphstatic.com/blog/3.31/sourcegraph-3-31-release.png
+socialImage: https://sourcegraphstatic.com/blog/3.31/sourcegraph-3-31-release.png
+changelogItems:
+  - description: 'Backend Code Insights GraphQL queries now support arguments `includeRepoRegex` and `excludeRepoRegex` to filter on repository names.'
+    url: https://github.com/sourcegraph/sourcegraph/pull/23256
+    category: Code Insights
+  - description: Operator documentation has been added to the Search Reference sidebar section.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23116
+    category: Search
+  - description: Syntax highlighting is now supported for CUE.
+    url: https://cuelang.org
+    category: Search
+  - description: Reintroduced a revised version of the Search Types sidebar section, making the quick links for search types (for example, diff searches) more intuitive to use.
+    url: https://github.com/sourcegraph/sourcegraph/pull/23170
+    category: Search
+  - description: "Added support for git p4's `--use-client-spec` option, allowing Sourcegraph to avoid cloning unimportant files. This can now be enabled by configuring the `p4.client` field in the Perforce integration."
+    url: https://github.com/sourcegraph/sourcegraph/pull/23845
+    category: Admin
```

Comment:
This is not relevant to admins or users. The important take away from this change is that our Perforce configuration now supports a field that enables admins to exclude files that aren't important.

```suggestion
  - description: "The Perforce integration has a new `p4.client` field that can be used to avoid cloning unimportant files by using git p4's `--use-client-spec`."
    url: https://github.com/sourcegraph/sourcegraph/pull/23845
    category: Admin
```